back/app/services/projects_finder_admin_service.rb

```diff
@@ -199,19 +199,16 @@ def self.filter_participation_states(scope, params = {})
 
   # Filter projects by the participation method of their current phase
   def self.filter_current_phase_participation_method(scope, params = {})
-    participation_methods = params[:participation_methods]
+    participation_methods = params[:participation_methods] || []
     return scope if participation_methods.blank?
 
-    participation_methods = Array(participation_methods)
+    current_phases_with_participation_methods = Phase
+      .where(participation_method: participation_methods)
+      .where("start_at <= current_date AND coalesce(end_at, 'infinity'::DATE) >= current_date")
 
-    project_ids = scope.pluck(:id)
-    project_ids_with_matching_phase = project_ids.select do |project_id|
-      project = Project.includes(:phases, :admin_publication).find_by(id: project_id)
-      next false unless project
+    project_ids_with_matching_phase = current_phases_with_participation_methods
```

Comment:
This way, we derive the project ids for the phases that we want with a single sub-query. In your solution, we need to loop over all projects, and for each project run a SQL query, and load the result into memory. This could become pretty slow for clients with many projects.